src/AnimatedFade.tsx

```diff
@@ -1,51 +1,103 @@
-import React, { useEffect, useRef } from 'react';
+import React, { useCallback, useEffect, useRef } from 'react';
 import { Animated, ViewStyle } from 'react-native';
+import useCurrent from './hooks/useCurrent';
 
 type Props = {
-  fadeOnMount?: boolean;
+  /**
+   * auto-trigger fade-in?
+   */
   triggerFade?: boolean;
-  startingOpacity?: number; // 0 - 1.00
-  endingOpacity?: number; // 0 - 1.00
-  duration?: number; // ms
-  delay?: number; // ms
+
+  /**
+   * starting opacity value (0 - 1.00)
+   */
+  opacityStart?: number;
+
+  /**
+   * ending opacity value (0 - 1.00)
+   */
+  opacityEnd?: number;
+
+  /**
+   * animation duration (in ms)
+   */
+  duration?: number;
+
+  /**
+   * delay before animation start (in ms)
+   */
+  delay?: number;
   style?: ViewStyle;
+
+  /**
+   * callback function called once fadeIn animation completes
+   */
   onEnd?: () => void;
+
+  /**
+   * should this animation repeat?
+   */
+  loop?: boolean;
 };
 
 const AnimatedFade: React.FC<Props> = ({
-  fadeOnMount = true,
-  triggerFade = false,
-  startingOpacity = 0,
-  endingOpacity = 1,
+  triggerFade = true,
+  opacityStart = 0,
+  opacityEnd = 1,
   duration = 800,
   delay = 0,
+  loop = false,
   onEnd,
   style,
   children,
 }) => {
-  const opacity = useRef(new Animated.Value(startingOpacity)).current;
+  const opacity = useRef(new Animated.Value(opacityStart));
+
+  const _onEnd = useCurrent(onEnd);
+  const handleEnd = useCallback(() => {
+    if (_onEnd.current) _onEnd.current();
+  }, [_onEnd]);
+
+  const fadeIn = () => {
+    Animated.timing(opacity.current, {
+      toValue: opacityEnd,
+      duration,
+      delay,
+      useNativeDriver: true,
+    }).start((status) => {
+      if (status.finished) {
+        handleEnd();
+        if (loop) fadeOut();
+      }
+    });
+  };
 
-  const fade = () => {
-    Animated.timing(opacity, {
-      toValue: endingOpacity,
+  const fadeOut = () => {
+    Animated.timing(opacity.current, {
+      toValue: opacityStart,
       duration,
       delay,
       useNativeDriver: true,
     }).start((status) => {
-      if (status.finished && onEnd) {
-        onEnd();
+      if (status.finished) {
+        handleEnd();
+        if (loop) fadeIn();
       }
     });
   };
 
+  const _fadeIn = useCurrent(fadeIn);
+
   useEffect(() => {
-    if (fadeOnMount || triggerFade) {
-      fade();
+    if (triggerFade) {
+      _fadeIn.current();
     }
-  }, [triggerFade, fadeOnMount]);
+  }, [triggerFade, _fadeIn]);
```

Comment:
Using `useCurrent` allows us to pass the ref instead of the `fadeIn` callback function into the useEffect dependency array, removing the possibility of an infinite loop.